nomad/structs/csi_volumes.go

```diff
@@ -0,0 +1,248 @@
+package structs
+
+import (
+	"fmt"
+	"strings"
+	"time"
+)
+
+const (
+	VolumeTypeCSI = "csi"
+)
+
+type CSIVolume struct {
+	ID         string
+	Driver     string
+	Namespace  string
+	Topology   *CSITopology
+	MaxReaders int
+	MaxWriters int
+
+	// Allocations, tracking claim status
+	ReadAllocs  map[string]*Allocation
+	WriteAllocs map[string]*Allocation
+	PastAllocs  map[string]*Allocation
+
+	// Healthy is true iff all the denormalized plugin health fields are true, and the
+	// volume has not been marked for garbage collection
+	Healthy           bool
+	VolumeGC          time.Time
+	Controller        *Job
```

Comment:
Linking to a Job might be difficult here, multiple nomad jobs might have the same CSI Controller tasks